packages/@expo/server/src/index.ts

```diff
@@ -126,6 +130,32 @@ export function createRequestHandler(
 
     debug('Request', sanitizedPathname);
 
+    // Execute middleware first, before any route matching
+    if (routesManifest.middleware) {
+      try {
+        // Check if middleware should run for this request based on matchers
+        if (shouldRunMiddleware(request, routesManifest.middleware)) {
+          const middlewareModule = await getMiddleware(routesManifest.middleware);
+          if (middlewareModule?.default) {
+            const shallowRequest = new ImmutableRequest(request);
+            const middlewareResponse = await middlewareModule.default(shallowRequest);
+            if (middlewareResponse instanceof Response) {
```

Comment:
We may need to add some `bodyUsed` checks here. If the user used `.clone()` they can still use the original body, which means that we receive the cloned, but used, `Request` clone here. We'd then want to switch back to the original request.